snapcraft/models/project.py

```diff
@@ -1336,16 +1334,30 @@ def validate_grammar(cls, data: dict[str, Any]) -> None:
 class ArchitectureProject(models.CraftBaseModel, extra="ignore"):
     """Project definition containing only architecture data."""
 
+    base: str = pydantic.Field(
+        description="The build environment to use when building the snap",
+        examples=["base: core20", "base: core22", "base: core24", "base: devel"],
+    )
+
     architectures: list[str | Architecture] = pydantic.Field(
-        default=[get_host_architecture()],
+        default=[DebianArchitecture.from_host()],
         validate_default=True,
     )
 
+    @pydantic.field_validator("base")
+    @classmethod
+    def _validate_base(
+        cls, value: str | None, info: pydantic.ValidationInfo
+    ) -> str | None:
+        """Build-base defaults to the base value if not specified."""
+        return value or info.data.get("base")
```

Comment:
I think this is a no-op because you have:
`@pydantic.field_validator("base")`
and then
`return value or info.data.get("base")`

One of those should be build-base, let me look at the code some more

Reply:
(outdated)